DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/bridging/test_bridging_learning_address.py

```diff
@@ -0,0 +1,143 @@
+        return
+    dent_dev = dent_devices[0]
+    device_host_name = dent_dev.host_name
+    tg_ports = tgen_dev.links_dict[device_host_name][0]
+    ports = tgen_dev.links_dict[device_host_name][1]
+    traffic_duration = 10
+
+    out = await IpLink.add(
+        input_data=[{device_host_name: [{"device": bridge, "type": "bridge"}]}])
+    assert out[0][device_host_name]["rc"] == 0,  f"Verify that bridge created.\n{out}"
+
+    out = await IpLink.set(
+        input_data=[{device_host_name:  [
+            {"device": port, "master": "br0", "operstate": "up"} for port in ports]},
+            {"device": bridge, "operstate": "up"}])
+    assert out[0][device_host_name]["rc"] == 0, f"Verify that bridge, bridge entities set to 'UP' state and links enslaved to bridge.\n{out}"
+    
+    out = await BridgeLink.set(
+        input_data=[{device_host_name: [
+            {"device": port, "learning": True, "flood": False} for port in ports]}])
+    assert out[0][device_host_name]["rc"] == 0, f"Verify that entities set to learning 'ON' and flooding 'OFF' state.\n{out}" 
+    
+    address_map = (
+        #swp port, tg port,     tg ip,      gw,         plen
+        (ports[0], tg_ports[0], "11.0.0.1", "11.0.0.4", 24),
+        (ports[1], tg_ports[1], "11.0.0.2", "11.0.0.3", 24),
+        (ports[2], tg_ports[2], "11.0.0.3", "11.0.0.2", 24),
+        (ports[3], tg_ports[3], "11.0.0.4", "11.0.0.1", 24),
+    )
+
+    dev_groups = tgen_utils_dev_groups_from_config(
+        {"ixp": port, "ip": ip, "gw": gw, "plen": plen}
+        for _, port, ip, gw, plen in address_map
+    )
+
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, ports, dev_groups)
+
+    streams = {
+        "bridge_1": {
+            "ip_source": dev_groups[tg_ports[3]][0]["name"],
+            "ip_destination": dev_groups[tg_ports[0]][0]["name"],
+            "srcMac": "aa:bb:cc:dd:ee:11",
+            "dstMac": "ff:ff:ff:ff:ff:ff",
+            "type": "raw",
+            "protocol": "802.1Q",
+        },
+        "bridge_2": {
+            "ip_source": dev_groups[tg_ports[2]][0]["name"],
+            "ip_destination": dev_groups[tg_ports[1]][0]["name"],
+            "srcMac": "aa:bb:cc:dd:ee:12",
+            "dstMac": "ff:ff:ff:ff:ff:ff",
+            "type": "raw",
+            "protocol": "802.1Q",
+        },
+        "bridge_3": {
+            "ip_source": dev_groups[tg_ports[1]][0]["name"],
+            "ip_destination": dev_groups[tg_ports[2]][0]["name"],
+            "srcMac": "aa:bb:cc:dd:ee:13",
+            "dstMac": "ff:ff:ff:ff:ff:ff",
+            "type": "raw",
+            "protocol": "802.1Q",
+        },
+        "bridge_4": {
+            "ip_source": dev_groups[tg_ports[0]][0]["name"],
+            "ip_destination": dev_groups[tg_ports[3]][0]["name"],
+            "srcMac": "aa:bb:cc:dd:ee:14",
+            "dstMac": "ff:ff:ff:ff:ff:ff",
+            "type": "raw",
+            "protocol": "802.1Q",
+        }
+    }
```

Comment:
You can use a dict comprehension here:
```python
streams = {
    f"bridge_{dst + 1}": {
        "ip_source": dev_groups[tg_ports[src]][0]["name"],
        "ip_destination": dev_groups[tg_ports[dst]][0]["name"],
        "srcMac": f"aa:bb:cc:dd:ee:1{dst + 1}",
        "dstMac": "ff:ff:ff:ff:ff:ff",
        "type": "raw",
        "protocol": "802.1Q",
    } for src, dst in ((3, 0), (2, 1), (1, 2), (0, 3))
}
```
Or even:
```python
streams = {
    f"bridge_{idx}": {
        "ip_source": dev_groups[src][0]["name"],
        "ip_destination": dev_groups[dst][0]["name"],
        "srcMac": f"aa:bb:cc:dd:ee:1{idx}",
        "dstMac": "ff:ff:ff:ff:ff:ff",
        "type": "raw",
        "protocol": "802.1Q",
    } for idx, (src, dst) in enumerate(zip(tg_ports, reversed(tg_ports)), start=1)
}
```
But the second example is way too unreadable, so maybe don't use it.